sql-plugin/src/main/java/com/nvidia/spark/rapids/iceberg/parquet/GpuParquet.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.iceberg.parquet;
+
+import java.util.Collection;
+import java.util.Map;
+
+import com.nvidia.spark.rapids.GpuMetric;
+import com.nvidia.spark.rapids.PartitionedFileUtils;
+import com.nvidia.spark.rapids.iceberg.data.GpuDeleteFilter;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.hadoop.HadoopInputFile;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.io.InputFile;
+import org.apache.iceberg.mapping.NameMapping;
+import org.apache.iceberg.relocated.com.google.common.collect.Sets;
+import org.apache.parquet.HadoopReadOptions;
+import org.apache.parquet.ParquetReadOptions;
+
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.execution.datasources.PartitionedFile;
+import org.apache.spark.sql.vectorized.ColumnarBatch;
+
+/** GPU version of Apache Iceberg's Parquet class */
```

Comment:
Yes, this ended up diverging more than most from the Iceberg class.  Iceberg uses a function because it supports batch vectorized vs. row-based readers, but we only have a single GPU reader so that was overkill.  As such there are things that normally only the function callback would need as context (such as the delete filter) that appears here instead.  Essentially I removed an unnecessary abstraction level here.